File: src/main/java/com/animalfarm/admin/collection/TradeHistory.java
package com.animalfarm.admin.collection;

import lombok.*;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Getter @Setter @ToString
@RequiredArgsConstructor
@Document("tradeHistory")
public class TradeHistory {
    @Id
    private String id;
    private Trader acceptor;
    private Trader requester;
    private String tradeDate;
}

File: src/main/kotlin/com/animalfarm/admin/collection/TradeHistory.kt
package com.animalfarm.admin.collection;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document

@Document("tradeHistory")
data class TradeHistory(
    @Id
    var id: String,
    var acceptor: Trader,
    var requester: Trader,
    var tradeDate: String
)

data class Trader(
    var uuid: String,
    var money: Long,
    var items: List<String>
)